zcash_client_backend/src/data_api/testing/pool.rs

```diff
@@ -4954,3 +4957,114 @@ pub fn receive_two_notes_with_same_value<T: ShieldedPoolTester>(
         assert_eq!(T::note_value(note.note()), value);
     }
 }
+
+fn build_coinbase_tx(
+    network: &LocalNetwork,
+    value: Zatoshis,
+    recipient: TransparentAddress,
+    miner_data: Option<PushValue>,
+) -> zcash_primitives::transaction::builder::BuildResult {
+    let build_config = BuildConfig::Coinbase { miner_data };
+    let mut builder = Builder::new(network.clone(), BlockHeight::from_u32(1), build_config);
+
+    // Add transparent output to recipient
+    builder.add_transparent_output(&recipient, value).unwrap();
+
+    // Build the transaction (coinbase transactions don't need provers)
+    builder
+        .build(
+            // unused internally
+            &TransparentSigningSet::new(),
+            // unused internally
+            &[],
+            // unused internally
+            &[],
+            OsRng,
+            &LocalTxProver::bundled(),
+            &LocalTxProver::bundled(),
+            // unused internally
+            &StandardFeeRule::Zip317,
+        )
+        .unwrap()
+}
+
+/// Tests that immature coinbase outputs are excluded from note selection.
+pub fn immature_coinbase_outputs_are_excluded_from_note_selection<T: ShieldedPoolTester>(
+    dsf: impl DataStoreFactory,
+    cache: impl TestCache,
+) {
+    let mut st = TestDsl::with_sapling_birthday_account(dsf, cache).build::<T>();
+
+    // Get the default transparent address
+    let (t_addr, _) = st.get_account().usk().default_transparent_address();
+
+    let coinbase_value = Zatoshis::const_from_u64(50000);
+
+    // Construct the coinbase transaction and mine the block
+    let coinbase_build_result = build_coinbase_tx(st.network(), coinbase_value, t_addr, None);
+    let coinbase_tx = coinbase_build_result.transaction();
+    let (h, _) = st.generate_next_block_from_tx(0, coinbase_tx);
+    st.scan_cached_blocks(h, 1);
+
+    // Store the UTXO in the wallet (not sure about this step)
```

Comment:
This is okay; another way to do this would be to call `decrypt_and_store_transaction`.